src/QsCompiler/CompilationManager/TypeChecking.cs

```diff
@@ -205,12 +204,13 @@ internal static ImmutableArray<string> DocumentingComments(this FileContentManag
         /// </summary>
         /// <param name="items">A collection of positioned items.</param>
         /// <exception cref="ArgumentException">No item precedes <paramref name="pos"/>.</exception>
-        private static T ContainingParent<T>(Position pos, IReadOnlyCollection<(Position, T)> items)
+        private static T ContainingParent<T>(Position pos, IReadOnlyCollection<(Position, T)> items, T? fallback = null)
+            where T : class
```

Comment:
I think it would be cleaner to return `T?` instead of taking in a default value.